source/custom/nifgen_service.custom.cpp

```diff
@@ -1,2 +1,64 @@
+#include <nifgen/nifgen_service.h>
+
+#include <stdexcept>
+
 namespace nifgen_grpc {
-}  // namespace nitclk_grpc
+
+class DriverErrorException : public std::runtime_error{
```

Comment:
You don't need `CheckStatus` and `DriverErrorException` yet. We only use it if we are making some other library calls before calling the main library API and want to return if those calls return error.

Reply:
Oh right. Removed it. Thanks Rohan